src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -896,6 +960,74 @@ class spell_dru_omen_of_clarity : public AuraScript
     }
 };
 
+// 274902 - Photosynthesis
+class spell_dru_photosynthesis : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_DRUID_LIFEBLOOM,
+            SPELL_DRUID_LIFEBLOOM_FINAL_HEAL,
+            SPELL_DRUID_LIFEBLOOM_OVERGROWTH
+        });
+    }
+
+    bool CheckProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        return roll_chance_i(aurEff->GetAmount()) &&
+            (eventInfo.GetActionTarget()->HasAura(SPELL_DRUID_LIFEBLOOM, eventInfo.GetActor()->GetGUID()) ||
+             eventInfo.GetActionTarget()->HasAura(SPELL_DRUID_LIFEBLOOM_OVERGROWTH, eventInfo.GetActor()->GetGUID()));
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        GetTarget()->CastSpell(eventInfo.GetActionTarget(), SPELL_DRUID_LIFEBLOOM_FINAL_HEAL, CastSpellExtraArgs(aurEff));
+    }
+
+    void Register() override
+    {
+        DoCheckEffectProc += AuraCheckEffectProcFn(spell_dru_photosynthesis::CheckProc, EFFECT_1, SPELL_AURA_DUMMY);
+        OnEffectProc += AuraEffectProcFn(spell_dru_photosynthesis::HandleProc, EFFECT_1, SPELL_AURA_DUMMY);
+    }
+};
+
+// 274906 - Photosynthesis
+// Called by Lifebloom and Lifebloom (Overgrowth)
+class spell_dru_photosynthesis_effect : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_PHOTOSYNTHESIS_EFFECT })
```

Comment:
I'm overall very confused about this script, i dont know what everything here is supposed to be